cfblog/models.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import unicode_literals
+
+__author__ = 'vinay'
+
+import datetime
+
+from django.conf import settings
+from django.db import models
+from django.core.exceptions import ValidationError
+from django.core.urlresolvers import resolve
+from django.utils.translation import ugettext_lazy as _
+
+from jsonfield import JSONField
+from tagging.fields import TagField
+
+from .managers import CmsBlogCategoryManager, CmsBlogPostManager
+from .utils import cacheable, stale_cache, get_public_data, get_image_path, get_file_path
+from .validators import validate_and_get_template, validate_url_path
+
+
+class CmsPageTemplate(models.Model):
+    name = models.CharField(_('template name'), max_length=255, db_index=True, validators=[validate_and_get_template])
+
+    def __unicode__(self):
+        return self.name
+
+
+STATUS_CHOICES = (
+    (1, _('Draft')),
+    (2, _('Public')),
+)
+
+
+class BaseCmsPage(models.Model):
+    template = models.ForeignKey(CmsPageTemplate, verbose_name=_('template'), null=True, blank=True)
+    auth_data = JSONField(verbose_name=_('author data'), default={}, blank=True)
+    public_data = JSONField(verbose_name=_('public data'), default={}, blank=True)
+    created_on = models.DateTimeField(_('created on'), auto_now_add=True)
+    modified_on = models.DateTimeField(_('modified on'), auto_now=True)
+    status = models.IntegerField(_('status'), choices=STATUS_CHOICES, default=1)
+    publish = models.DateTimeField(_('publish'), default=datetime.datetime.now)
+
+    class Meta:
+        abstract = True
+
+    @property
+    def is_public(self):
+        return self.status == 2
+
+    def publish_cms_content(self):
+        self.public_data = self.auth_data
+        self.status = 2
+        self.save()
+
+
+class CmsPage(BaseCmsPage):
```

Comment:
We won't have this model. There is only one model call Content, which has both page and post. There is no difference between page and post. 

Also since the models.py is already for your stuff, you do not need to put Cms before everything. Table name cfblog_content is better than cfblog_cmscontent.